### PR TITLE
juju deploy create/overwrite charm version file

### PR DESCRIPTION
**Pull request by vinu2003**
## Description of change
Why is this change needed?
This PR has code changes for juju deploy command creating or overwriting charm version file for local charm if it is in revision control directory.
## QA steps
How do we verify that the change works?
deploy a local charm. 
(clone the charm from git repo inot local directoy first and deploy it.)
## Documentation changes
Does it affect current user workflow? CLI? API?
juju deploy command needs to be updated.
## Bug reference
Does this change fix a bug? Please add a link to it.
N/A

### CODE REVIEW

**Comment by vinu2003** [comment]
Moving the version file verification function to charm.v6 folder so that it can be used by charm-push command as well.

File: apiserver/charms.go [hunk 1]
```diff
@@ -419,6 +419,9 @@ func (h *charmsHandler) repackageAndUploadCharm(st *state.State, archive *charm.
 			errors.New("invalid version file")
 			version = ""
 		}
+		if len(version) > 100 {
```
Comment (wallyworld) [inline apiserver/charms.go]: can do this in one line fmt.Sprintf("%.100s", version)

File: api/client.go [hunk 2]
```diff
@@ -327,6 +331,10 @@ func (c *Client) AddLocalCharm(curl *charm.URL, ch charm.Charm) (*charm.URL, err
 		}
 	case *charm.CharmArchive:
 		var err error
+		// For local charm check if it is revision control directory
```
Comment (wallyworld) [inline api/client.go]: Comment can be expanded a bit  - we're generating a version file not just checking it is a vcs dir

Comment (wallyworld) [reply api/client.go]: Also, a charm archive is a zip file. There's no need to generate a version file for that case

File: api/client_test.go [hunk 3]
```diff
@@ -40,6 +41,7 @@ import (
 	"github.com/juju/juju/testcharms"
 	coretesting "github.com/juju/juju/testing"
 	"github.com/juju/juju/testing/factory"
+	"github.com/juju/utils"
```
Comment (wallyworld) [inline api/client_test.go]: move me

**Comment by vinu2003** [comment]
$$merge$$

**Pull request merged**